lib/iex/lib/iex/evaluator.ex

```diff
@@ -58,17 +58,49 @@ defmodule IEx.Evaluator do
   # https://github.com/elixir-lang/elixir/issues/1089 for discussion.
   @break_trigger "#iex:break\n"
 
+  @space " "
+  @leading_binary_operators ["not in", "or", "and", "in"] ++ ~w(
```

Comment:
This can lead to false positives if a variable is named `orion`. You will need to check if those are followed by a space. :)

Reply:
Followed by a space is easy, we might simply put a space into the pattern itself. But `:foo in[:foo]` is somehow valid syntax, as well as `true or+([:foo])`. I surely could check if the next symbol is an allowed-for-variable-name Unicode character, but I am not sure if we need to be as captious here.

So I personally would go with
```suggestion
  @leading_binary_operators ["not in ", "or ", "and ", "in "] ++ ~w(
```

Thoughts?

Reply:
I see. Then we are reimplementing the parser considerably which is a bit too much. I have plans to rewrite the parser so we could actually return this information more easily but this is quite some time ahead...

Reply:
I think a reasonable compromise for now is to support this only for the pipe operators in the .yrl file. WDYT?

Reply:
Well, the main purpose of this PR is still to enable copy-paste blocks as is. The more I think the more hardly could I see why one would do `++ [:foo]` (and if needed, there is still `v() ++ [:foo]`.) So I would accept only what `Code.format_string!/2` welcomes at the beginning of the line. That might be tested as:

1. produce a bunch of snippets with the operator on another line
2. run `mix format` on it
3. see what goes where (I am almost sure only pipe operators are not pulled up to the previous line, but I have to check)
```

Reply:
I like the formatter idea. You can find the list of operators that can go on the next line in lib/code/formatter.ex: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/lib/code/formatter.ex#L21:L25

Reply:
Actually, because when is only allowed in patterns, we don’t need to include it. There is probably no need to include | either, so just go with the first module attribute!

Reply:
Maybe it’d make sense to internally export (`@doc false`) this list from `lib/code/formatter.ex` to borrow it here, but because of the aforementioned parser rewrite, I think it’s good enough as it is, at least for now.